doc/sphinx/language/core/definitions.rst

```diff
@@ -1,5 +1,22 @@
-Definitions
-===========
+Definitions and theorems
+========================
+
+Definitions associate a specified term with a given name. The name can later
+be replaced with its definition through :term:`δ-reduction <delta-reduction>`.
```

Comment:
```suggestion

Definitions and theorems are used to define terms that have a specific type.

Definitions most commonly associate a specified term with a given name.
Alternatively, the type may be specified instead of the term, in which case, a proof
must be provided to derive the term just as for theorems. The name can later
be replaced with its definition through :term:`δ-reduction <delta-reduction>`.
```

The first sentence suggests they are similar.  Details of their differences should go later
where `Definition` syntax is given (e.g. opacity and that specifying both term and type
will verify that they match).  N'est-ce pas?